### ISSUE TITLE
Cancel compute server edit

### ISSUE THREAD

**Issue by novoselt** [sagemathinc/cocalc#7461]
There is a button "Save" when editing a compute server, but really its meaning is "Exit" because as soon as I change something it is saved and clicking X in the corner is the same as clicking "Save" as far as I can tell.

On the other hand, if I experimented with some settings, I may want to just close the configurator WITHOUT changing anything, i.e. I want to discard any edits and keep everything as it was. So it would be nice to have a button "Discard changes" or something like that!

**Comment by williamstein** [comment]
I agree.  My plan is that things work like with Admin Settings, where you can make several changes, they get clearly indicated, and you can explicitly save one (or all of them) or revert them.  That's just not what ended up getting implemented so far though, since it's a lot more work.   I should temporarily change "Save" to "Exit" or "Close", then change it back when it's properly implemented!

**Comment by williamstein** [comment]
Labeling this as "hard" since it's a massive refactor and lots of potential subtle issues, across different implementations.  Also, the value is not so clear.
